chia/cmds/farm_funcs.py

```diff
@@ -89,7 +90,14 @@ async def summary(
     root_path: Path = DEFAULT_ROOT_PATH,
 ) -> None:
     harvesters_summary = await get_harvesters_summary(farmer_rpc_port, root_path)
-    blockchain_state = await get_blockchain_state(rpc_port, root_path)
+    blockchain_state = None
+    try:
+        blockchain_state = await get_blockchain_state(rpc_port, root_path)
+    except CliRpcConnectionError:
+        pass
+    except Exception as ex:
+        print(ex)
```

Comment:
Converting an exception to a string gives only the text, not the type or the traceback.  In cases like `some_dict[a_not_present_key]` this gives you just a string of the key.  Some exceptions don't even bother having messages at all and would just be blank.

```suggestion
    except Exception as e:
        print(traceback.print_exc(file=sys.stderr))
```

I think there could be some refinement around this, including applying similar handling to other such requests in this function, but having the exception at the top of the output and on stderr still leaves the rest of the output intact, including on redirect to files or piping to other programs etc.

You may have to add imports for `sys` and `traceback`.

Thanks again.

Reply:
My bad, got the exceptions mixed up. The connection error is wrapped in a `CliRpcConnectionError` which always provides a nice message. But this is of course precisely _not_ handling the `CliRpcConnectionError` :)

Fixed. And I also adjusted the exception handling for the similar local wallet connection.

Reply:
Thanks! Got the exceptions mixed up. The connection error is wrapped in a CliRpcConnectionError which always provides a nice message. But this is of course precisely not handling the CliRpcConnectionError :)

Fixed. And I also adjusted the exception handling for the similar local wallet connection.